js/original.js

```diff
@@ -143,6 +145,70 @@
   function calculateScore() {
     score += Planets.length * Stars.length * 5;
     if (Planets.length == 0) score -= 1;
+    if (score > nextLevel && score < nextLevel + (nextLevel / 200)) {
+      levelUp();
+    }
+  }
+  function levelUp() {
+    let sunLocationX = randomInt(radius, innerWidth - radius);
+    let sunLocationY = randomInt(radius + 100, innerHeight - radius);
+    let existingX = [];
+    let existingY = [];
+    let distance = 20; // Default max distance new stars should have from existing ones
+
+    Stars.forEach(function(star) {
+      existingX.push(star.x);
+      existingY.push(star.y);
+    });
+
+    let i = 0;
+
+    while (i < existingX.length) {
+      for (var z = 0; z < existingX.length; z++) {
+        if (!between(sunLocationX, existingX[z] - distance, // If not within 20px of any existing x co-ords
```

Comment:
For detecting collision, we can use formula from geometry:-
If two circles collide then:-
Distance between them <=2*r

This code is implemented here (not complete copy though) https://github.com/nveenjain/gravity/pull/10/files#diff-51771aa99a43d67cef66bdda737d4fe0R33